administrator/components/com_media/controllers/file.json.php

```diff
@@ -56,6 +56,7 @@ public function upload()
 		// Get some data from the request
 		$file   = $this->input->files->get('files', '', 'array');
 		$folder = $this->input->get('folder', '', 'path');
+		$canOverwrite = $this->input->getBool('allowovewrite', false);
```

Comment:
This whole plugin needs a major cleanup anyway and I started part of it but not sure who is going to do what now. Are you taking over?